vos/vos.py

```diff
@@ -1452,22 +1461,33 @@ def copy(self, source, destination, send_md5=False):
                     response = self.conn.session.get(get_url, timeout=(2, 5), stream=True)
                     source_md5 = response.headers.get('Content-MD5', source_md5)
                     response.raise_for_status()
-                    with open(destination, 'w') as fout:
+                    # Write to a temporary file instead of destination.
+                    if destination[0:4] == 'vos:':
+                        tobj = NamedTemporaryFile(prefix='vospace.')
```

Comment:
Ok, I've had time to take a closer look. You're right about the delete flag - sorry I have no experience with NamedTemporaryFile and I misread the docs. As for when the file is closed, it is recommended to be used, as any normal file handler in Python, in conjunction with either try-finally or with so that the file handler is not carried over by a thrown exception. Other than that it looks good.